Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -887,16 +887,12 @@ RefPtr<Statement> Parser::try_parse_labelled_statement(AllowLabelledFunction all
         if (function_declaration->kind() == FunctionKind::Async)
             syntax_error("Async functions cannot be defined in labelled statements");
 
-        labelled_statement = move(function_declaration);
+        labelled_item = move(function_declaration);
     } else {
         m_state.labels_in_scope.set(identifier, {});
-        labelled_statement = parse_statement(allow_function);
-        if (is<IterationStatement>(*labelled_statement)) {
+        labelled_item = parse_statement(allow_function);
+        if (is<IterationStatement>(*labelled_item))
```

Comment:
This (or really with the LabelledStatement change) breaks the check below for continue statement in multi labelled loops.
As the failing test says the following should parse:
```js
outer: inner: for (var i = 0; i < 5; i++)
   continue inner; // or continue outer;
```
but
```js
outer: inner: {
     continue inner; // parse error cannot continue a block
}
```
should fail. 
Now to be honest the current check is kind of fragile.
Since this is quite rare we can probably just keep checking the statement until we hit a non LabelledStatement:
Which might have to a lot but I don't think double labels really occur in practice.

Reply:
Fixed :)